apps/src/applab/setPropertyDropdown.js

```diff
@@ -167,16 +172,22 @@ PROP_NAMES[ElementType.SLIDER] = [
 ];
 
 // Create a mapping of PROPS_PER_TYPE[elementType][friendlyName] => prop info
+// and a mapping of PROP_NAMES_PER_TYPE[elementType] => array of properties
 var PROPS_PER_TYPE = {};
+var PROP_NAMES_PER_TYPE = {};
 Object.keys(PROP_NAMES).map(function (elementType) {
   PROPS_PER_TYPE[elementType] = {};
+  PROP_NAMES_PER_TYPE[elementType] = [];
   PROP_NAMES[elementType].forEach(function (propName) {
```

Comment:
Having `PROP_NAMES`, `PROPS_PER_TYPE` and now `PROP_NAMES_PER_TYPE`, all keyed on `elementType`, seems like an antipattern.  Can we do this with fewer data structures... or at least give them names that differentiate them better?  I'm thinking either `PROPS_PER_TYPE[elementType][friendlyName]` gets changed to an array, or `PROP_NAMES_PER_TYPE` is renamed to `FRIENDLY_NAMES_PER_TYPE`?  Or am I misunderstanding here?